### DISCUSSION TITLE
MapLibre support WMS of type vector ?

### DISCUSSION THREAD

**Discussion by ambarja** [maplibre/maplibre-gl-js#2065] [Q&A]
Good afternoon, is it possible to consume services from a geoserver? 
I have seen that it is possible with matrix data, but I don't know how to implement it when they are vector data or when they are in WMS format.
Can someone help me ? 
Greetings!
**Vector** : 
 - http://mtcgeo2.mtc.gob.pe:8080/geoserver/MTC_pg/red_vial_nacional_dic18/wms?service=WMS&request=GetCapabilities&layers=MTC_pg:red_vial_nacional_dic18
 
**Raster**:
 - https://www.idep.gob.pe/geoportal/rest/directories/arcgisoutput/IMAGENES/ORTOFOTO_AREQUIPA_MapServer/_ags_mapf727c9afc59648ed9bb72120def6f25b.png

**Comment by HarelM** [comment]
The above two URLs are not reachable for me, not sure why though.
In any case, I'm not sure this is currently supported, but I think it might be supported using the `addProtocol` method.
You'll need to convert the data to mvt or geojson but I think this is possible although I'm not familiar with the format.
There has been a few plugins and tutorials about using the add protocol method to support exotic formats and other stuff :-)

**Comment by tomass** [comment]
Just for information: WMS - Web Map Service - is a service which returns you an image (png, jpeg), not vector data (see part "service=WMS" in first link). You provide an extent, layers, filter, maybe even styling info and it returns you an image. Therefore it is a raster service, not vector.

**Reply by HarelM** [reply]
Oh, I thought you are talking about something else.
Yes, this is already supported, here's a code example I use in my site to support this:
https://github.com/IsraelHikingMap/Site/blob/4b069880db64b76e16708c83a3b12de651b9faba/IsraelHiking.Web/src/application/components/map/automatic-layer-presentation.component.ts#L132
